aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -0,0 +1,254 @@
+    /// schedule derived from these parameter will be for gas calculation of
+    /// the entire next epoch.
+    /// -- The data is one epoch stale than ideal, but VM doesn't need to worry
+    /// about reloading gas parameters after the first txn of an epoch.
+    struct StorageGas has key {
+        per_item_read: u64,
+        per_item_create: u64,
+        per_item_write: u64,
+        per_byte_read: u64,
+        per_byte_create: u64,
+        per_byte_write: u64,
+    }
+
+    // x and y are basis points.
+    struct Point has copy, drop, store {
+        x: u64,
+        y: u64
+    }
+
+    /// P(x) = min_price + (base ^ (utilization / target_utilization) - 1) / (base - 1) * (max_price - min_price)
+    // initialize the base to be 32, which means:
+    //   When DB is at 50% target utilization,the price increases roughly 15% of (max_price - min_price) on top of min_price.
+    //   More data points:
+    //     10% -> 1%
+    //     20% -> 3%
+    //     30% -> 6%
+    //     40% -> 10%
+    //     50% -> 15%
+    //     60% -> 23%
+    //     70% -> 33%
+    //     80% -> 48%
+    //     90% -> 70%
+    //     95% -> 84%
+    //     99% -> 96%
+    //    100% -> 100%
+    struct UsageGasConfig has copy, drop, store {
+        min_gas: u64,
+        max_gas: u64,
+        target_utilization: u64,
+        read_curve: vector<Point>,
+        create_curve: vector<Point>,
+        write_curve: vector<Point>,
+    }
+
+    struct StorageGasConfig has copy, drop, key {
+        item_config: UsageGasConfig,
+        byte_config: UsageGasConfig,
+    }
+
+    public fun new_curve_point(x: u64, y: u64): Point {
+        Point {x, y}
+    }
+
+    public fun new_usage_gas_config(min_gas: u64, max_gas: u64, target_utilization: u64, read_curve: vector<Point>, create_curve: vector<Point>, write_curve: vector<Point>): UsageGasConfig {
+        UsageGasConfig {
+            min_gas,
+            max_gas,
+            target_utilization,
+            read_curve,
+            create_curve,
+            write_curve,
+        }
+    }
+
+    public fun new_storage_gas_config(item_config: UsageGasConfig, byte_config: UsageGasConfig): StorageGasConfig {
+        StorageGasConfig {
+            item_config,
+            byte_config
+        }
+    }
+
+    public(friend) fun set_config(aptos_framework: &signer, config: StorageGasConfig) acquires StorageGasConfig {
+        system_addresses::assert_aptos_framework(aptos_framework);
+        validate_usage_config(&config.item_config);
+        validate_usage_config(&config.item_config);
+        *borrow_global_mut<StorageGasConfig>(@aptos_framework) = config;
+    }
+
+    public(friend) fun initialize(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+        assert!(
+            !exists<StorageGasConfig>(@aptos_framework),
+            error::already_exists(ESTORAGE_GAS_CONFIG)
+        );
+        let standard_curve: vector<Point> = vector[
```

Comment:
This is where you should document the curve?

Maybe a method `10_point_base_32_exponential_curve() -> Curve` and document there.